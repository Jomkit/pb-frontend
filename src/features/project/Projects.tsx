import { useContext, useEffect, useState } from 'react'
import Frame from '../../components/Frame'
import ProjectList from './ProjectList'
import Api from '../../api'
import userContext from '../../components/contexts/userContext'

const Projects = () => {
  const initialState = [
    {
      id: "1",  
      name: "Example project",
      note: "Notes about example project"
    }
  ]

  const [projects, setProjects] = useState(initialState);
  const [isLoading, setIsLoading] = useState(true);
  const {id} = useContext(userContext);
  const userId = id;
  
  useEffect(() => {
    const getProjects = async () => {
      if(!userId) return;
      const projects = await Api.getUserProjects(userId);
      
      if(projects.length !== 0){
        setProjects(projects);
      }
      setIsLoading(false);
    }
    getProjects();
  }, [userId]);
  
  return (
    <Frame bgColor='bg-cyan-400' title='Projects'>
      { isLoading ? <p>Loading...</p> : <ProjectList projects={projects}/> }
    </Frame>
  )
}

export default Projects