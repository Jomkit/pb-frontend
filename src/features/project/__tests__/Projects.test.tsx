import { render } from "@testing-library/react"
import { MemoryRouter } from "react-router-dom"
import Projects from "../Projects"

it("should render without crashing", () => {
    render(
        <MemoryRouter>
            <Projects />
        </MemoryRouter>
    )
})

it("should match snapshot while loading projects", () => {
    const { asFragment } = render(
        <MemoryRouter>
            <Projects />
        </MemoryRouter>
    )
    expect(asFragment()).toMatchSnapshot();
})

it("should match snapshot after loading complete", async () => {
    const { asFragment, findByText } = render(
        <MemoryRouter>
            <Projects />
        </MemoryRouter>
    )
    setTimeout(async () => {
        let testProjectText = await findByText("testProject 1");
        expect(testProjectText).toBeInTheDocument();
        expect(asFragment()).toMatchSnapshot();
    }, 200);
})