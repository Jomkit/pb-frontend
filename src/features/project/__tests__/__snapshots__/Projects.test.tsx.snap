// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`should match snapshot while loading projects 1`] = `
<DocumentFragment>
  <section
    class="text-center items-start min-h-screen w-screen p-4 bg-cyan-400"
  >
    <h1>
      Projects
    </h1>
    <div
      class="p-10 justify-center"
    >
      <p>
        Loading...
      </p>
    </div>
  </section>
</DocumentFragment>
`;
